src/server.py

```diff
@@ -52,6 +71,21 @@ def __bus_message_cb(self, unused_bus, message):
                             debug)
             self.pipeline.set_state(Gst.State.NULL)
             self.emit("error", error, debug)
+        elif message.type == Gst.MessageType.STATE_CHANGED:
+            if message.src != self.pipeline:
+                return
+            st = message.get_structure()
+            old_state = st.get_value("old-state")
+            new_state = st.get_value("new-state")
+            if old_state == Gst.State.READY and new_state == Gst.State.PAUSED:
+                if self.speed is not None:
```

Comment:
I'd suggest moving this up, before `message.get_structure()` and doing an early return:
```py
if self.speed is None:
    return
```